examples/05_advanced/plot_ica_neurovault.py

```diff
@@ -0,0 +1,143 @@
+"""
+NeuroVault cross-study ICA maps.
+================================
+
+This example shows how to download statistical maps from
+NeuroVault, label them with NeuroSynth terms,
+and compute ICA components across all the maps.
+
+See :func:`nilearn.datasets.fetch_neurovault`
+documentation for more details.
+
+"""
+# Author: Ben Cipollini
+# License: BSD
+# Ported from code authored by Chris Filo Gorgolewski, Gael Varoquaux
+# https://github.com/NeuroVault/neurovault_analysis
+import warnings
+
+import numpy as np
+from scipy import stats
+from sklearn.decomposition import FastICA
+
+from nilearn.datasets import fetch_neurovault
+from nilearn.image import smooth_img
+
+from nilearn.datasets import load_mni152_brain_mask
+from nilearn.input_data import NiftiMasker
+
+from nilearn import plotting
+import matplotlib.pyplot as plt
+
+
+######################################################################
+# Get image and term data
+# -----------------------
+
+# Download images
+# Here by default we only download 80 images to save time,
+# but for better results I recommend using at least 200.
+print("Fetching Neurovault images; "
+      "if you haven't downloaded any Neurovault data before "
+      "this will take several minutes.")
+nv_data = fetch_neurovault(max_images=80, fetch_neurosynth_words=True)
+
+images = nv_data['images']
+term_weights = nv_data['word_frequencies']
+vocabulary = nv_data['vocabulary']
+
+# Clean and report term scores
+term_weights[term_weights < 0] = 0
+total_scores = np.mean(term_weights, axis=0)
+
+print("\nTop 10 neurosynth terms from downloaded images:\n")
+
+for term_idx in np.argsort(total_scores)[-10:][::-1]:
+    print(vocabulary[term_idx])
+
+
+######################################################################
+# Reshape and mask images
+# -----------------------
+
+print("\nReshaping and masking images.\n")
+
+with warnings.catch_warnings():
+    warnings.simplefilter('ignore', UserWarning)
+    warnings.simplefilter('ignore', DeprecationWarning)
+
+    mask_img = load_mni152_brain_mask()
+    masker = NiftiMasker(
+        mask_img=mask_img, memory='nilearn_cache', memory_level=1)
+    masker = masker.fit()
+
+    # Images may fail to be transformed, and are of different shapes,
+    # so we need to transform one-by-one and keep track of failures.
+    X = []
+    is_usable = np.ones((len(images),), dtype=bool)
+
+    for index, image_path in enumerate(images):
+        # load image and remove nan and inf values.
```

Comment:
Comment is misleading isn't it removing nan and inf ?

Why don't we smooth with NiftiMasker ?